src/operator/src/statement/ddl.rs

```diff
@@ -746,24 +754,98 @@ fn find_partition_entries(
+        for partition in &partitions.exprs {
+            let partition_expr = convert_one_expr(partition, &column_name_and_type)?;
+            partition_exprs.push(vec![PartitionBound::Expr(partition_expr)]);
+        }
+
+        // fallback for no expr
+        if partition_exprs.is_empty() {
+            partition_exprs.push(vec![PartitionBound::MaxValue]);
+        }
+
+        partition_exprs
     } else {
         vec![vec![PartitionBound::MaxValue]]
     };
     Ok(entries)
 }
 
+fn convert_one_expr(
+    expr: &Expr,
+    column_name_and_type: &HashMap<&String, ConcreteDataType>,
+) -> Result<PartitionExpr> {
+    let Expr::BinaryOp { left, op, right } = expr else {
+        return InvalidPartitionRuleSnafu {
+            reason: "partition rule must be a binary expression",
+        }
+        .fail();
+    };
+
+    let op =
+        RestrictedOp::try_from_parser(&op.clone()).with_context(|| InvalidPartitionRuleSnafu {
+            reason: format!("unsupported operator in partition expr {op}"),
+        })?;
+
+    // convert leaf node.
+    let (lhs, op, rhs) = match (left.as_ref(), right.as_ref()) {
+        (Expr::Identifier(ident), Expr::Value(value)) => {
+            let (column_name, data_type) = convert_identifier(ident, column_name_and_type)?;
+            let value = convert_value(value, data_type)?;
+            (Operand::Column(column_name), op, Operand::Value(value))
+        }
+        (Expr::Value(value), Expr::Identifier(ident)) => {
+            let (column_name, data_type) = convert_identifier(ident, column_name_and_type)?;
+            let value = convert_value(value, data_type)?;
+            (Operand::Value(value), op, Operand::Column(column_name))
+        }
+        (Expr::BinaryOp { .. }, Expr::BinaryOp { .. }) => {
+            // sub-expr must against another sub-expr
+            let lhs = convert_one_expr(left, column_name_and_type)?;
+            let rhs = convert_one_expr(right, column_name_and_type)?;
+            (Operand::Expr(lhs), op, Operand::Expr(rhs))
+        }
+        _ => {
+            return InvalidPartitionRuleSnafu {
+                reason: format!("invalid partition expr {expr}"),
+            }
+            .fail();
+        }
+    };
+
+    Ok(PartitionExpr::new(lhs, op, rhs))
+}
+
+fn convert_identifier(
+    ident: &Ident,
+    column_name_and_type: &HashMap<&String, ConcreteDataType>,
+) -> Result<(String, ConcreteDataType)> {
+    let column_name = ident.value.clone();
+    let data_type = column_name_and_type
+        .get(&column_name)
+        .cloned()
+        .with_context(|| ColumnNotFoundSnafu { msg: &column_name })?;
+    Ok((column_name, data_type))
+}
+
+fn convert_value(value: &ParserValue, data_type: ConcreteDataType) -> Result<Value> {
+    // TODO(ruihang): pass timezone here
+    sql_value_to_value("<NONAME>", &data_type, value, None).context(ParseSqlValueSnafu)
+}
+
+/// Merge table level table options with schema level table options.
```

Comment:
Looks like we don't get back the test `test_parse_partitions` in this mod.

Reply:
>This relates to selecting regions to query. We currently returns every region.

So that test won't pass currently